chessboard/gallen/chessboard.go

```diff
@@ -0,0 +1,45 @@
+package chessboard
+
+const RankSize = 8
+
+type Rank [RankSize]bool
+
+type Chessboard map[string]Rank
+
+func CountInRank(cb Chessboard, rank string) (count int) {
+    for _, occupied := range cb[rank] {
+        if (occupied) {
+            count++
+        }
+    }
+    return
+}
+
+func CountInFile(cb Chessboard, file int) (count int) {
+    if (file <= RankSize) {
```

Comment:
Negative check, I missed that.

As for the earlier return, that's just a readability thing I assume.  I guess I agree, but for some reason, I prefer seeing it fall through to the end. I do think multiple return spots can lead to confusion at times.  But for one like this I can see where it might be preferable.  Old habits die hard.

Reply:
Yeah, "one single return" has been my mantra since forever. But I see how this becomes a problem sometimes, making the code more difficult to read.

In this specific case, I'm not even sure. Both options are fine for me.